packages/firestore/src/model/field_value.ts

```diff
@@ -235,6 +254,22 @@ export class DoubleValue extends NumberValue {
     }
   }
 
+  toProto(useProto3Json: boolean): api.Value {
+    if (useProto3Json) {
+      // Proto 3 lets us encode NaN and Infinity as string values as
```

Comment:
This comment reads backwards, as if it's describing the proto format, when really it's the JSON encoding of the proto that has this string form.  In other words, both encodings are proto 3 so we really need to distinguish between JSON or binary. It would be clearer if you wrote it something like this:

> In the JSON encoding, NaN and Infinity are string values ...